src/librustc_typeck/check/mod.rs

```diff
@@ -3858,13 +3858,9 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                           element_ty
                       }
                       None => {
-                          let mut err = self.type_error_struct(
-                              expr.span,
-                              |actual| {
-                                  format!("cannot index a value of type `{}`",
-                                          actual)
-                              },
-                              base_t);
+                          let mut err = struct_span_err!(tcx.sess, expr.span, E0608,
+                                                         "cannot index a value of type `{}`",
```

Comment:
This is separate than the `Into` trait stuff. I've always heard indexing described as being into something, like you "index into an array", you don't "index an array". The latter sounds weird to me. With the current PR you use "index into" in the detailed explanation, but the short error title is still "index a value". This should at least be consistent.